### ISSUE TITLE
Error when running grunt with extract option

### ISSUE THREAD

**Issue by MartinRixham** [jscs-dev/node-jscs#1949]
I have the following jscs config in my Gruntfile

```
jscs: {

    src: ["src/*.js", "tests/*.html"],
    options: {

        config: ".jscsrc",
        esnext: false,
        verbose: true,
        fix: false,
        extract: true
    }
}
```

but i get a parse error for each of my html files

```
parseError: Unexpected token < at tests/value.html :
     1 |<!DOCTYPE html>
---------^
     2 |<html>
     3 | <head
```

`npm list jscs` gives

```
└─┬ grunt-jscs@2.3.0
  └── jscs@2.5.0 
```

I have tried this on fedora, ubuntu and mac.

**Comment by markelog** [comment]
The issue is not with the grunt plugin, but with logic of `extract` property - 
1. It can't process file if it is not in the directory i.e; this - 
`jscs index.html --extract *.html` will not work
2. It can't correctly process directory files too i.e; this - 
`jscs tmp --extract *.html` (if there is file `index.html` inside `tmp` directory) will not work, only 
`jscs index.html --extract *.html` this which pretty weird considering that you don't have `index.html` file in the root, only inside `tmp` folder.

In other words, `extract` requires a lot more work and is not really in working condition at the moment.

/cc @lahmatiy

**Comment by lahmatiy** [comment]
You should specify html file masks in `extract` option. But for now `extract` option check file names that in `src` subtrees. I'm not sure, but this might work:

```js
jscs: {
    src: ["src/*.js", "tests"],
    options: {
        config: ".jscsrc",
        esnext: false,
        verbose: true,
        fix: false,
        extract: ["tests/*.html"]
    }
}
```

@markelog Yep, `extract` option is a little bit confuse. But it's by design, since it's an additional for regular checking. If you want to check exact file, you should use something like this:

```
jscs *.* --extract index.html
```

But you couldn't disable default file checking (i.e. `.js` files), since no cli option for that.

Original issue is not connected with `extract` itself. It's about no check for `fileExtensions` when match for filename. In other words `.html` file is processing as `.js` because its filename matches to `src` pattern, but shouldn't as extension isn't in `fileExtensions` list.

I thought about future feature extension. Believe it should be more extensible, predictable, that applies for any files which extension not in `fileExtensions` list, with nesting support (for example, js in markdown that in jsdocs that in js that in html file). And as we can see, there is should be some way to set some path for extracting and avoid regular `js` file checking.

**Comment by MartinRixham** [comment]
Ok that works. Thanks for the help.
